controllers/user.controller.js

```diff
@@ -0,0 +1,54 @@
+/* eslint-disable no-underscore-dangle */
+const User = require('../models/user.model');
+
+// update user
+const updateUser = async (ctx, next) => {
+  try {
+    const user = await User.findByIdAndUpdate(
+      { _id: ctx.request.body._id },
+      ctx.request.body,
+      { new: true },
+    );
+    // send back updated user
+    ctx.body = user;
+  } catch (error) {
+    ctx.status = error.statusCode || error.status || 500;
+    ctx.body = { message: error.message };
+  }
+  await next();
+};
+
+// check if password is correct and if user exists
+const checkUser = async (ctx, next) => {
```

Comment:
I see what is the idea behind this controller function for the sign-in endpoint, but remember how to implement a Basic authentication with a get request, where the username and password should be passed in the authorization header.